api/envoy/extensions/http/original_ip_detection/custom_header/v3/custom_header.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+package envoy.extensions.http.original_ip_detection.custom_header.v3;
+
+import "envoy/type/v3/http_status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.http.original_ip_detection.custom_header.v3";
+option java_outer_classname = "CustomHeaderProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Custom header original IP detection extension]
+
+// This extension allows for the original downstream remote IP to be detected
+// by reading the value from a configured header name. If the value is successfully parsed
+// as an IP, it'll be treated as the effective downstream remote address and seen as such
+// by all filters. See :ref:`original_ip_detection_extensions
+// <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.original_ip_detection_extensions>`
+// for an overview of how extensions operate and what happens when an extension fails
+// to detect the remote IP.
+//
+// [#extension: envoy.http.original_ip_detection.custom_header]
+message CustomHeaderConfig {
+  // Configuration options for rejecting requests when detection fails.
+  message RejectOptions {
+    // If reject_request_if_detection_fails is set, this sets the HTTP status that is sent downstream
+    // when detection fails. The default status is HTTP 403 Forbidden.
+    type.v3.HttpStatus status_on_error = 1;
+
+    // If reject_request_if_detection_fails is set, this sets the HTTP body in the response
+    // when detection fails. If not set, an empty body is used.
+    string body_on_error = 2;
+  }
+
+  // The header name containing the original downstream remote address, if present.
+  //
+  // Note: in the case of a multi-valued header, only the first value is tried and the rest are ignored.
+  string header_name = 1
+      [(validate.rules).string = {min_len: 1 well_known_regex: HTTP_HEADER_NAME strict: true}];
```

Comment:
Hmm yeah I think this should be `strict: false` to match all the other examples, so I'm confused what issue you ran into. Can you chat with @asraa about this as I think she is most familiar? I'm not completely sure what issue you ran into but maybe your example in the test was not using a character that is blocked in strict false mode?

Reply:
Hmmm so strict: true will actually enforce HTTP header validation, while strict: false will only block what Envoy does normally for `HeaderString` (\r\0\n). What was the issue?

Reply:
Ah yes, that'd explain why my test for invalid headers wasn't working (I wasn't using \r\0\n).

Interesting though that we use strict: false in all proto files... meaning we are pretty relaxed when checking header names. 

Shall I stick to that convention or can we go with strict: true? Strict: true is better, except that it's inconsistent with what's done for every other config...

Reply:
It's kinda up to you! The reason we didn't was because we couldn't make API validations more strict with the annotation.

Since this is new, why not. (especially if you will be enforcing this in the code anyway)

Reply:
Ah gotcha. Yeah it'd would make sense then. Thanks!